Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/TrackFindingAlgorithm.hpp

```diff
@@ -125,56 +130,41 @@ void TrackFindingAlgorithm::computeSharedHits(
-    auto& measIndexes = ckfResult.lastMeasurementIndices;
-
-    for (auto measIndex : measIndexes) {
-      ckfResult.fittedStates->visitBackwards(measIndex, [&](const auto& state) {
-        if (not state.typeFlags().test(Acts::TrackStateFlag::MeasurementFlag)) {
-          return true;
-        }
-
-        std::size_t hitIndex =
-            state.uncalibrated().template get<IndexSourceLink>().index();
-
-        // Check if hit not already used
-        if (firstTrackOnTheHit.at(hitIndex) ==
-            std::numeric_limits<std::size_t>::max()) {
-          firstTrackOnTheHit.at(hitIndex) = iresult;
-          firstStateOnTheHit.at(hitIndex) = state.index();
-          return true;
-        }
-
-        // if already used, control if first track state has been marked
-        // as shared
-        int indexFirstTrack = firstTrackOnTheHit.at(hitIndex);
-        int indexFirstState = firstStateOnTheHit.at(hitIndex);
-        if (not results.at(indexFirstTrack)
-                    .value()
-                    .fittedStates->getTrackState(indexFirstState)
-                    .typeFlags()
-                    .test(Acts::TrackStateFlag::SharedHitFlag)) {
-          results.at(indexFirstTrack)
-              .value()
-              .fittedStates->getTrackState(indexFirstState)
-              .typeFlags()
-              .set(Acts::TrackStateFlag::SharedHitFlag);
-        }
-
-        // Decorate this track
-        results.at(iresult)
-            .value()
-            .fittedStates->getTrackState(state.index())
-            .typeFlags()
-            .set(Acts::TrackStateFlag::SharedHitFlag);
-
-        return true;
-      });
+  for (auto track : tracks) {
+    for (auto state : track.trackStates()) {
+      if (not state.typeFlags().test(Acts::TrackStateFlag::MeasurementFlag)) {
+        break;
+      }
+
+      std::size_t hitIndex =
+          state.uncalibrated().template get<IndexSourceLink>().index();
+
+      // Check if hit not already used
+      if (firstTrackOnTheHit.at(hitIndex) ==
+          std::numeric_limits<std::size_t>::max()) {
+        firstTrackOnTheHit.at(hitIndex) = track.index();
+        firstStateOnTheHit.at(hitIndex) = state.index();
+        break;
+      }
+
+      // if already used, control if first track state has been marked
+      // as shared
+      int indexFirstTrack = firstTrackOnTheHit.at(hitIndex);
+      int indexFirstState = firstStateOnTheHit.at(hitIndex);
+
+      auto firstState = tracks.getTrack(indexFirstTrack)
+                            .container()
+                            .trackStateContainer()
+                            .getTrackState(indexFirstState);
+      if (not firstState.typeFlags().test(
+              Acts::TrackStateFlag::SharedHitFlag)) {
+        firstState.typeFlags().set(Acts::TrackStateFlag::SharedHitFlag);
+      }
+
+      // Decorate this track
+      state.typeFlags().set(Acts::TrackStateFlag::SharedHitFlag);
+
+      break;
```

Comment:
```suggestion
      continue;
```
or remove completely.